python/whylogs/core/model_performance_metrics/confusion_matrix.py

```diff
@@ -92,12 +92,55 @@ def add(
         if not isinstance(predictions, list):
             predictions = [predictions]
 
-        if scores is None:
-            scores = [1.0 for _ in range(len(targets))]
-
         if len(targets) != len(predictions):
             raise ValueError("both targets and predictions need to have the same length")
 
+        if len(self.labels) > 2:
+            _multiclass = True
+        else:
+            _multiclass = False
+
+        if scores is None:
+            if _multiclass:
+                score_row = [1.0] * len(self.labels)
+                scores = [score_row] * len(targets)
+            else:
+                scores = [1.0] * len(targets)
+
+        if _multiclass:
+            if isinstance(scores[0], list):
+                print("received multidimensional scores object")
+                if any(length != len(self.labels) for length in [len(row) for row in scores]):
+                    raise ValueError("scores shape must be (n, d) for n rows and d unique labels")
+                # TODO: Until capable of handling new message format, convert scores
+                # to a single list with only predicted class. This will not result in
+                # the correct ROC and score-related values, but confusion matrix accurate.
+                scores = [row[row.index(max(row))] for row in scores]
```

Comment:
if for now this leads to incorrect ROC and score-related results, show we warn the user for this case also? I'm not sure if the ROC and score-related values is calculated in whylabs. If so, maybe the warning would be better placed there?

Might be missing context here, just speculating.